src/main/java/mops/persistence/FileRepository.java

```diff
@@ -0,0 +1,176 @@
+     *
+     * @param configuration the injected Config.
+     * @throws StorageException on Error
+     */
+    public FileRepository(FileRepositoryConfig configuration) throws StorageException {
+        this.configuration = configuration;
+        try {
+            this.minioClient = new MinioClient(
+                    configuration.getHost(),
+                    configuration.getPort(),
+                    configuration.getAccessKey(),
+                    configuration.getSecretKey()
+            );
+        } catch (InvalidEndpointException | InvalidPortException e) {
+            throw new StorageException(e);
+        }
+
+        try {
+            if (!minioClient.bucketExists(configuration.getBucketName())) {
+                minioClient.makeBucket(configuration.getBucketName());
+            }
+        } catch (InvalidBucketNameException | NoSuchAlgorithmException | InsufficientDataException
+                | IOException | InvalidKeyException | NoResponseException | XmlPullParserException
+                | ErrorResponseException | InternalException | InvalidResponseException | RegionConflictException e) {
+            throw new StorageException(e);
+        }
+    }
+
+    /**
+     * Saves a file.
+     *
+     * @param file   the file that should get saved permanently.
+     * @param fileId the File ID given by the FileInfo database.
+     * @throws StorageException on Error
+     */
+    public void saveFile(MultipartFile file, long fileId) throws StorageException {
+        try (InputStream stream = file.getInputStream()) {
+            minioClient.putObject(configuration.getBucketName(),
+                    String.valueOf(fileId),
+                    stream,
+                    file.getSize(),
+                    new HashMap<>(),
+                    null, // no encryption will be needed
+                    file.getContentType()
+            );
+        } catch (MinioException | IOException | InvalidKeyException
+                | NoSuchAlgorithmException | XmlPullParserException e) {
+            throw new StorageException(e);
+        }
+    }
+
+    /**
+     * Deletes a file permanently.
+     *
+     * @param fileId the ID of the file that's desired to be deleted.
+     * @return true if successful; false if not.
+     * @throws StorageException on error
+     */
+    public boolean deleteFile(long fileId) throws StorageException {
+        try {
+            minioClient.removeObject(
+                    configuration.getBucketName(),
+                    String.valueOf(fileId)
+            );
+        } catch (InvalidBucketNameException | NoSuchAlgorithmException | InsufficientDataException
+                | IOException | InvalidKeyException | NoResponseException | XmlPullParserException
+                | ErrorResponseException | InternalException | InvalidArgumentException | InvalidResponseException e) {
+            throw new StorageException(e);
+        }
+
+        return !fileExist(fileId);
```

Comment:
is that additional request even necessary?
and do we even want to return a boolean here?

Reply:
Good point. ErrorResponseException will be thrown on unsuccessful execution. So this will always return true or throw an exception.